src/celestia/celx_category.cpp

```diff
@@ -0,0 +1,99 @@
+
+#include <fmt/printf.h>
+#include "celx.h"
+#include "celx_internal.h"
+#include "celx_category.h" 
+#include "celx_object.h"
+#include <celengine/category.h>
+#include <celengine/selection.h>
+
+static int category_tostring(lua_State *l)
+{
+    CelxLua celx(l);
+
+    UserCategory *c = *celx.getThis<UserCategory*>();
+    std::string s = fmt::sprintf("[UserCategory:%s]", c->name());
+    return celx.push(s.c_str());
+}
+
+static int category_getname(lua_State *l)
+{
+    CelxLua celx(l);
+
+    UserCategory *c = *celx.getThis<UserCategory*>();
+    const char *n = c->name().c_str();
+    return celx.push(n);
+}
+
+static int category_createchild(lua_State *l)
+{
+    CelxLua celx(l);
+
+    UserCategory *c = *celx.getThis<UserCategory*>();
+    const char *n = celx.safeGetString(2);
```

Comment:
check for nullptr

Reply:
I added additional check inside Celx_safeGetString.

Reply:
I did not understood it, I'll create a new issue for tis case. I just need to make some checks before this.

Reply:
```
category:createchild()
-- or
category:createchild(1)
```
and celestia is aborted

because you did not check for nullptr